test/ItemCategoriesServiceTest.php

```diff
@@ -0,0 +1,137 @@
+        $itemProphecy1 = $this->prophesize('\core_kernel_classes_Resource');
+        $itemProphecy2 = $this->prophesize('\core_kernel_classes_Resource');
+        $itemProphecy3 = $this->prophesize('\core_kernel_classes_Resource');
+        $itemProphecy4 = $this->prophesize('\core_kernel_classes_Resource');
+        $itemProphecy5 = $this->prophesize('\core_kernel_classes_Resource');
+
+        $properties = array('subject', 'difficulty');
+
+        $item1Properties = array(
+            'subject' => array(
+                new \core_kernel_classes_Resource('science'),
+                new \core_kernel_classes_Literal('math')
+            ),
+            'difficulty' => array(
+                new \core_kernel_classes_Literal('medium')
+
+            )
+        );
+
+        $item2Properties = array(
+            'subject' => array(
+                new \core_kernel_classes_Resource('ELA'),
+            )
+        );
+
+        $item3Properties = array(
+            'subject' => array(
+                new \core_kernel_classes_Literal('math')
+            ),
+            'difficulty' => array(
+                new \core_kernel_classes_Literal('easy')
+
+            )
+        );
+
+        $item4Properties = array();
+
+        $item5Properties = array(
+            'subject' => array(
+                new \core_kernel_classes_Resource('ELA'),
+            ),
+            'difficulty' => array(
+                new \core_kernel_classes_Literal('hard')
+
+            )
+        );
+
+        $itemProphecy1->getPropertiesValues($properties)->willReturn($item1Properties);
+        $itemProphecy1->getUri()->willReturn('itemUri1');
+        $itemProphecy2->getPropertiesValues($properties)->willReturn($item2Properties);
+        $itemProphecy2->getUri()->willReturn('itemUri2');
+        $itemProphecy3->getPropertiesValues($properties)->willReturn($item3Properties);
+        $itemProphecy3->getUri()->willReturn('itemUri3');
+        $itemProphecy4->getPropertiesValues($properties)->willReturn($item4Properties);
+        $itemProphecy4->getUri()->willReturn('itemUri4');
+        $itemProphecy5->getPropertiesValues($properties)->willReturn($item5Properties);
+        $itemProphecy5->getUri()->willReturn('itemUri5');
+
+        $items = array(
+            $itemProphecy1->reveal(),
+            $itemProphecy5->reveal(),
+            $itemProphecy3->reveal(),
+            $itemProphecy4->reveal(),
+            $itemProphecy2->reveal(),
+        );
+
+        $expected = array(
+            'itemUri1' => array('SCIENCE', 'MATH', 'MEDIUM'),
+            'itemUri5' => array('ENGLISH', 'HIGH'),
+            'itemUri3' => array('MATH', 'LOW'),
+            'itemUri4' => array(),
+            'itemUri2' => array('ENGLISH'),
+        );
+        /** @var ItemCategoriesService $itemCategoriesService */
+        $itemCategoriesService = new ItemCategoriesService(
+            array(
+                'properties'    => array('subject', 'difficulty'),
```

Comment:
I would rather suggest to bind a mapping to a single property rather than multiple, to improve isolation and possible side effects on large mappings.

e.g. 

```php
array(
    'http://mysite.com#prop1' =>
    array(
        'mysite.com#instance1' => 'MATH',
        'français' => 'FRENCH'
    ),
    'http://mysite.com#prop2' => 
    array(
        'doutchlund' => 'GERMAN'
    )
)
```